python_modules/dagster/dagster_tests/cli_tests/command_tests/test_materialize_command.py

```diff
@@ -122,6 +131,54 @@ def test_conflicting_partitions():
         )
 
 
+def test_partition_and_partition_range_options():
+    with instance_for_test():
+        result = invoke_materialize(
+            "daily_partitioned_asset",
+            partition="2020-01-01",
+            partition_range_start="2020-01-01",
+        )
+        assert (
+            "Cannot specify both --partition and --partition-range options. Use only one."
+            in str(result.exception)
+        )
+
+
+def test_partition_range_end_missing():
+    with instance_for_test():
+        result = invoke_materialize(
+            "daily_partitioned_asset",
+            partition_range_start="2020-01-01",
+        )
+        assert "Both --partition-range-start and --partition-range-end must be provided." in str(
+            result.exception
+        )
+
+
+def test_partition_range_start_missing():
+    with instance_for_test():
+        result = invoke_materialize(
+            "daily_partitioned_asset",
+            partition_range_end="2020-01-01",
+        )
+        assert "Both --partition-range-start and --partition-range-end must be provided." in str(
+            result.exception
+        )
+
+
+def test_partition_range_single_run_backfill_policy():
+    with instance_for_test() as instance:
+        result = invoke_materialize(
+            "daily_partitioned_asset",
+            partition_range_start="2020-01-01",
+            partition_range_end="2020-01-03",
+        )
+        assert "RUN_SUCCESS" in result.output
+        event = instance.get_latest_materialization_event(AssetKey("daily_partitioned_asset"))
+        assert event is not None
+        assert event.asset_materialization.partition == "2020-01-03"
```

Comment:
Since you're backfilling three days of data here, we should validate that all three days are materialized. You could use e.g. `instance.get_materialized_partitions(...)`